DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,31 @@
+## How It Works
+
+The project itself is a derivative of [finstyle](https://github.com/fnichol/finstyle), but starts with all cops (rules) disabled. Cookstyle has a direct dependency on one specific version of RuboCop (at a time), and patches it to load the upstream configuration and custom set of cop updates. When a new RuboCop release comes out, this library can bump the pinned RuboCop version dependency and [re-vendor](https://github.com/chef/cookstyle/blob/master/Rakefile) the upstream configuration to determine if any breaking style or lint cops were added/dropped/reversed/etc.
+
+## NOTE CAREFULLY ABOUT UPDATING COOKSTYLE
+
+Cookstyle is designed to allow bumping the RuboCop engine while keeping backwards compatibility with the config and not requiring all the cookbooks to update. This isn't *always* possible since many bugfixes catch new issues in cookbooks, but we do our best to prevent constantly linting of large codebases.
+
+Enabling or disabling any new cops in this file must be accompanied by comments demonstrating why the change should be made. Keep in mind anything you add here causes Chef users significant work, so make sure it's worth it.
+
+## Cookstyle Configuration Files
+
+Cookstyle works by overwriting RuboCop's standard config with our own default.yml file. This file sources three other files to provide the final layer-cake that is Cookstyle:
+
+[upstream.yml](https://github.com/chef/cookstyle/blob/master/config/upstream.yml)
+
+This is a vendored copy of the out-of-the-box RuboCop config from the version of the engine we're using. It's necessary to start with this file as it includes cop metadata like descriptions and also includes the actual configuration details for each cop.
```

Comment:
What is "This"? Can you please specify if it is "This `upstream.yml`" or some other file?

Reply:
fixed